vllm/model_executor/layers/linear.py

```diff
@@ -776,6 +931,19 @@ def weight_loader(self, param: Parameter, loaded_weight: torch.Tensor):
         assert param_data.shape == loaded_weight.shape
         param_data.copy_(loaded_weight)
 
+    def weight_loader_v2(self, param: vLLMParameter,
+                         loaded_weight: torch.Tensor):
+
+        if isinstance(param, ROW_PARALLEL_LOAD_SUPPORTED):
```

Comment:
Could `load_row_parallel_weight` just be a no-op instead of checking `ROW_PARALLEL_LOAD_SUPPORTED`

Reply:
yes - tradeoff would be more code to the parameter classes